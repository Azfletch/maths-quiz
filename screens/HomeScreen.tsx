import { Text, View, Image, Pressable } from 'react-native'
import React from 'react'
import { useNavigation } from '@react-navigation/native'

const HomeScreen = () => {
  const navigation = useNavigation()

  return (
    <View style={{ marginTop: 30 }}>
      <Image
        source={{ uri: 'https://freesvg.org/img/Math-Girl-.png' }}
        style={{ height: 370, width: '100%', resizeMode: 'contain' }}
      />

      <View style={{ padding: 10 }}>
        <Text style={{ textAlign: 'center', color: '#d257d2', fontWeight: '600', fontSize: 20 }}>QUIZ RULES</Text>
        <View style={{ padding: 10, borderRadius: 6, backgroundColor: '#f2f6a2', marginTop: 15 }}>
          <View style={{ alignItems: 'flex-start' }}>
            <View style={{ flexDirection: 'row', alignItems: 'center', margin: 5 }}>
              <Text style={{ color: '#d257d2', fontWeight: '500' }}>•</Text>
              <Text style={{ marginLeft: 5, color: '#d257d2', fontWeight: '500' }}>Each Question will display a 10 frame</Text>
            </View>
            <View style={{ flexDirection: 'row', alignItems: 'center', margin: 5 }}>
              <Text style={{ color: '#d257d2', fontWeight: '500' }}>•</Text>
              <Text style={{ marginLeft: 5, color: '#d257d2', fontWeight: '500' }}>Type Your Answer in the input provided</Text>
            </View>
            <View style={{ flexDirection: 'row', alignItems: 'center', margin: 5 }}>
              <Text style={{ color: '#d257d2', fontWeight: '500' }}>•</Text>
              <Text style={{ marginLeft: 5, color: '#d257d2', fontWeight: '500' }}>Answer all 5 questions to see your score!</Text>
            </View>
          </View>
        </View>
      </View>
      <Pressable
        style={{ backgroundColor: '#d057d2', padding: 15 , borderRadius: 15, marginLeft: 'auto', marginRight: 'auto', marginTop: 15 }}
        onPress={() => navigation.navigate('Quiz')}
      >
        <Text style={{ color: 'white', fontWeight: '600', fontSize: 20 }}>START QUIZ</Text>
      </Pressable>
    </View>
  )
}

export default HomeScreen