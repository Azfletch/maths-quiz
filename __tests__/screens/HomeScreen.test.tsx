
import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'

import HomeScreen from '../../screens/HomeScreen'

describe('HomeScreen', () => {
  it('navigates to the quiz on button press', () => {
    // Arrange
    const navigate = jest.fn()

    // Act
    const homeScreen = render(<HomeScreen navigation={{ navigate }} />)
    fireEvent.press(homeScreen.getByText('START QUIZ'))

    // Assert
    expect(homeScreen).toMatchSnapshot()
    expect(navigate).toHaveBeenCalledWith('Quiz')
  })
})  