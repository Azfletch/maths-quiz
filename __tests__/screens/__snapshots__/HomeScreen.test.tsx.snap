// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen navigates to the quiz on button press 1`] = `
<View
  style={
    {
      "marginTop": 30,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://freesvg.org/img/Math-Girl-.png",
      }
    }
    style={
      {
        "height": 370,
        "resizeMode": "contain",
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "#d257d2",
          "fontSize": 20,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      QUIZ RULES
    </Text>
    <View
      style={
        {
          "backgroundColor": "#f2f6a2",
          "borderRadius": 6,
          "marginTop": 15,
          "padding": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "flex-start",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "margin": 5,
            }
          }
        >
          <Text
            style={
              {
                "color": "#d257d2",
                "fontWeight": "500",
              }
            }
          >
            •
          </Text>
          <Text
            style={
              {
                "color": "#d257d2",
                "fontWeight": "500",
                "marginLeft": 5,
              }
            }
          >
            Each Question will display a 10 frame
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "margin": 5,
            }
          }
        >
          <Text
            style={
              {
                "color": "#d257d2",
                "fontWeight": "500",
              }
            }
          >
            •
          </Text>
          <Text
            style={
              {
                "color": "#d257d2",
                "fontWeight": "500",
                "marginLeft": 5,
              }
            }
          >
            Type Your Answer in the input provided
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "margin": 5,
            }
          }
        >
          <Text
            style={
              {
                "color": "#d257d2",
                "fontWeight": "500",
              }
            }
          >
            •
          </Text>
          <Text
            style={
              {
                "color": "#d257d2",
                "fontWeight": "500",
                "marginLeft": 5,
              }
            }
          >
            Answer all 5 questions to see your score!
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#d057d2",
        "borderRadius": 15,
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": 15,
        "padding": 15,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "600",
        }
      }
    >
      START QUIZ
    </Text>
  </View>
</View>
`;
